src/core/models.py

```diff
@@ -171,19 +173,34 @@ def password_rotated(self) -> bool:
         return bool(self.relation_data.get("password-rotated", None))
 
     @property
-    def hostname(self) -> str:
+    def hostname(self) -> str | None:
         """The hostname for the unit."""
-        return socket.gethostname()
+        if not self.fqdn:
+            return None
+        return self.fqdn.split(".")[0]
 
     @property
-    def fqdn(self) -> str:
+    def fqdn(self) -> str | None:
         """The Fully Qualified Domain Name for the unit."""
-        return socket.getfqdn()
+        if not (hostnames := socket.gethostbyaddr(self.private_ip)):
+            return None
+        return hostnames[0]
 
     @property
     def private_ip(self) -> str:
         """The IP for the unit."""
-        return socket.gethostbyname(self.hostname)
+        # Either values are okay for us because we are interested on their name
+        # both objects have a .name property.
+        address = None
+        if binding := self.relation:
+            binding_obj = self.model.get_binding(binding.name)
+            address = (
+                str(binding_obj.network.bind_address)
+                if binding_obj and binding_obj.network
```

Comment:
when can any of these 2 be ever `None`?

Reply:
@Mehdi-Bendriss it is because OSD has a way more strict linting tool. It essentially panics if it cannot resolve a given as a type or `None` (in this case, `binding_obj.network`). This was just a way around to tell the lint we will never have `None` on the subsequent line...

Personally, it is not really useful but I could not find a better way. Happy to hear any alternative

Reply:
Actually, `binding_obj.network.bind_address` can be `None` if you have no binding set. The `binding_obj.network` issue is that `lint` cannot validate its type, hence it raises a failure as unknown type or None. That is why I am adding this second part of the check.

Reply:
This is not the case anymore.